stdlib/args.ysh

```diff
@@ -129,27 +140,42 @@ func parseArgs(spec, argv) {
   var rest = []
 
   var value
+  var found
   while (i < argc) {
     var arg = argv[i]
     if (arg->startsWith('-')) {
+      setvar found = false
+
       for flag in (spec.flags) {
         if ( (flag.short and flag.short === arg) or
              (flag.long and flag.long === arg) ) {
           case (flag.type) {
             ('bool') | (null) { setvar value = true }
             int {
               setvar i += 1
-              setvar value = int(argv[i])
+              if (i >= len(argv)) {
+                error "Expected integer after '$arg'" (status=2)
```

Comment:
I think `2` is probably right, but again it makes me wonder if this API is ergonomic for users

We also have that issue when you do `case (_status) { (2) { echo } }` to some degree